### PR TITLE
V4L2 driver updates

### PR DESCRIPTION
**Pull request by 6by9**
A few updates to the V4L2 driver for the Pi Camera.
- Correct the support for ISO control.
- Add option to specify the /dev/videoN node number.
- Support for multiple cameras on the CM.
- Support for specifying where the overlay is on the screen.
- Compliance test failure.

A quick review by others would be appreciated, just to make sure I haven't missed something silly in the init/exit loops. (I'm afraid I haven't done as much testing of that as I perhaps ought to have done)

### CODE REVIEW

File: drivers/media/platform/bcm2835/bcm2835-camera.c [hunk 1]
```diff
@@ -54,6 +54,10 @@ int bcm2835_v4l2_debug;
 module_param_named(debug, bcm2835_v4l2_debug, int, 0644);
 MODULE_PARM_DESC(bcm2835_v4l2_debug, "Debug level 0-2");
 
+int video_nr = -1;
```
Comment (popcornmix) [inline drivers/media/platform/bcm2835/bcm2835-camera.c]: static?

File: drivers/media/platform/bcm2835/bcm2835-camera.c [hunk 2]
```diff
@@ -54,9 +56,10 @@ int bcm2835_v4l2_debug;
 module_param_named(debug, bcm2835_v4l2_debug, int, 0644);
 MODULE_PARM_DESC(bcm2835_v4l2_debug, "Debug level 0-2");
 
-int video_nr = -1;
-module_param_named(video_nr, video_nr, int, 0644);
-MODULE_PARM_DESC(video_nr, "videoX start number, -1 is autodetect");
+#define UNSET (-1)
+static int video_nr[] = {[0 ... (MAX_BCM2835_CAMERAS - 1)] = UNSET };
```
Comment (popcornmix) [inline drivers/media/platform/bcm2835/bcm2835-camera.c:60]: Ah - static!

File: drivers/media/platform/bcm2835/bcm2835-camera.c [hunk 3]
```diff
@@ -1792,8 +1895,10 @@ static int __init bm2835_mmal_init(void)
 free_dev:
 	kfree(dev);
 
-	v4l2_err(&dev->v4l2_dev,
-		 "%s: error %d while loading driver\n",
+	for ( ; camera > 0; camera--)
+		bcm2835_cleanup_instance(gdev[camera]);
```
Comment (popcornmix) [inline drivers/media/platform/bcm2835/bcm2835-camera.c:1935]: Should this cleanup gdev[0]?
Should you set gdev[camera] = NULL; ?

Comment (6by9) [inline drivers/media/platform/bcm2835/bcm2835-camera.c:1935]: Possibly. 
This is failure to load, so won't that unload the module due to the failure and therefore deallocate gdev.

Probably safest to reset it to NULL anyway - I'll update it.

File: drivers/media/platform/bcm2835/bcm2835-camera.c (hunk 1 above)
Comment (6by9) [inline drivers/media/platform/bcm2835/bcm2835-camera.c]: Perhaps same argument should be applied to bcm2835_v4l2_debug above.  As you've seen, I've copy/pasted from somewhere else in the next patch and it has become static :-)

**Comment by 6by9** [comment]
@popcornmix I think that addresses the things you raised, and adds an extra patch making the module parameters static variables, except bcm2835_v4l2_debug which is used in multiple files).

**Pull request merged**